### PR TITLE
#1947 Test Harness should support Light and Dark Mode.

### PR DESCRIPTION
**Pull request by srikant-ch5**
Fixes: #1947 

Uses Carbon Theme to switch between light and dark mode. https://react.carbondesignsystem.com/?path=/docs/components-theme--overview
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by matthoward366** [comment]
With this change do we even need https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/assets/styles/harness.scss#L22 anymore?

Does this work in production mode?

**Comment by srikant-ch5** [comment]
> With this change do we even need https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/assets/styles/harness.scss#L22 anymore?

We wont need above css with this change

> 
> Does this work in production mode?  

yes this is working in production mode. 


But the only issue I am facing is `Modal` and `Tearsheet` since these uses `Portals` which exist outside of DOM hierarchy but since Carbon Component's Theme will be applied only to children in this case Theme is not applied to Modal and Tearsheet. If we import through `scss` then theme is applied to both.

**Comment by srikant-ch5** [comment]
> > With this change do we even need https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/assets/styles/harness.scss#L22 anymore?
> 
> We wont need above css with this change
> 
> > Does this work in production mode?
> 
> yes this is working in production mode.
> 
> But the only issue I am facing in `Modal` and `Tearsheet` since these use `Portals` which exist outside of DOM hierarchy but since Carbon Component's Theme will be applied only to children in this case Theme is not applied to Modal and Tearsheet. If we import through `scss` then theme is applied to both.



> > With this change do we even need https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/assets/styles/harness.scss#L22 anymore?
> 
> We wont need above css with this change
> 
> > Does this work in production mode?
> 
> yes this is working in production mode.
> 
> But the only issue I am facing in `Modal` and `Tearsheet` since these use `Portals` which exist outside of DOM hierarchy but since Carbon Component's Theme will be applied only to children in this case Theme is not applied to Modal and Tearsheet. If we import through `scss` then theme is applied to both.

Hi @matthoward366 above issue is resolved, please review.

File: canvas_modules/common-canvas/src/common-properties/components/properties-modal/properties-modal.jsx [hunk 1]
```diff
@@ -40,21 +41,23 @@ export default class PropertiesModal extends Component {
...
-					onRequestSubmit={this.props.okHandler}
-					onSecondarySubmit={this.props.cancelHandler}
-					aria-label=""
-					size={modalSize}
-				>
-					<div className="properties-modal-children">
-						{this.props.children}
-					</div>
-				</Modal>
+				<Theme theme={this.props.light ? "g10" : "g90"}>
```
Comment (matthoward366) [inline canvas_modules/common-canvas/src/common-properties/components/properties-modal/properties-modal.jsx:44]: This doesn't seem correct.  What is a consumer wants a different theme?

**Comment by srikant-ch5** [comment]
Hi @matthoward366 I have included theme changes using `harness.scss` file now `Modal` and `Tearsheet` Components are also changing on theme change, please review.

File: canvas_modules/harness/assets/styles/harness.scss [hunk 2]
```diff
@@ -17,15 +17,23 @@
...
 // Modify common-canvas/src/themes/light.scss to switch common-canvas.
 $theme: $g10; // The light (gray 10) theme
 // $theme: $g90; // The dark theme
 
 :root {
-	@include theme();
+	@include theme($g10);
+}
+
+[data-carbon-theme='g90'] {
```
Comment (matthoward366) [inline canvas_modules/harness/assets/styles/harness.scss:30]: These both should have double quotes.  It's going to fail the build.

Comment (srikant-ch5) [reply canvas_modules/harness/assets/styles/harness.scss]: Yes, fixed these.

**Review by matthoward366 — COMMENTED** [review]
Looks good.  Can you fix this error?

Screenshot 2024-05-21 at 6 29 56 AM

**Comment by srikant-ch5** [comment]
> Looks good. Can you fix this error?
> 
> Screenshot 2024-05-21 at 6 29 56 AM

Pushed the change regarding this.

**Review by matthoward366 — APPROVED** [review]
lgtm

**Comment by srikant-ch5** [comment]
Hi @tomlyn Could you please review this ?

**Comment by tomlyn** [comment]
@srikant-ch5 Looks like Matt beat me to it.

**Pull request merged**